pennylane/ops/functions/map_wires.py

```diff
@@ -103,7 +103,7 @@ def map_wires(
         measurements = [qml.map_wires(m, wire_map) for m in input.measurements]
         prep = [qml.map_wires(p, wire_map) for p in input._prep]  # pylint: disable=protected-access
 
-        return QuantumScript(ops=ops, measurements=measurements, prep=prep)
+        return input.__class__(ops=ops, measurements=measurements, prep=prep)
```

Comment:
if they input an inherited class (aka QuantumTape), return a QuantumTape (not a QuantumScript)

Reply:
Thanks for the precision